el-maven-plugin/src/main/java/org/eclipse/persistence/build/embedmongo/StartMojo.java

```diff
@@ -0,0 +1,90 @@
+/*
+ * Copyright (c) 2023 Oracle and/or its affiliates. All rights reserved.
+ *
+ * This program and the accompanying materials are made available under the
+ * terms of the Eclipse Public License v. 2.0 which is available at
+ * http://www.eclipse.org/legal/epl-2.0,
+ * or the Eclipse Distribution License v. 1.0 which is available at
+ * http://www.eclipse.org/org/documents/edl-v10.php.
+ *
+ * SPDX-License-Identifier: EPL-2.0 OR BSD-3-Clause
+ */
+
+// Contributors:
+//     Oracle - initial API and implementation
+package org.eclipse.persistence.build.embedmongo;
+
+import de.flapdoodle.embed.mongo.config.Net;
+import de.flapdoodle.embed.mongo.transitions.Mongod;
+import de.flapdoodle.embed.mongo.transitions.RunningMongodProcess;
+import de.flapdoodle.embed.process.config.DownloadConfig;
+import de.flapdoodle.embed.process.io.ProcessOutput;
+import de.flapdoodle.embed.process.net.HttpProxyFactory;
+import de.flapdoodle.embed.process.net.ProxyFactory;
+import de.flapdoodle.embed.process.transitions.DownloadPackage;
+import de.flapdoodle.reverse.TransitionWalker;
+import de.flapdoodle.reverse.transitions.Start;
+
+import org.apache.maven.plugin.MojoExecutionException;
+import org.apache.maven.plugin.MojoFailureException;
+import org.apache.maven.plugins.annotations.LifecyclePhase;
+import org.apache.maven.plugins.annotations.Mojo;
+import org.apache.maven.settings.Settings;
+
+import java.net.URI;
+
+import static org.apache.commons.lang3.StringUtils.equalsIgnoreCase;
+import static org.codehaus.plexus.util.StringUtils.contains;
+
+/**
+ * When invoked, this goal starts an instance of mongo. The required binaries
+ * are downloaded if no mongo release is found in <code>~/.embedmongo</code>.
+ *
+ */
+@Mojo(name="start-mongo", defaultPhase = LifecyclePhase.PRE_INTEGRATION_TEST)
```

Comment:
is this threadsafe? take a look at parallel builds

Reply:
It's not threadsafe. Added explicit `, threadSafe = false` . Maybe not needed as `org.apache.maven.plugins.annotations.Mojo` has `boolean threadSafe() default false;`